template.yml

```diff
@@ -851,54 +852,57 @@ Resources:
               - logs:CreateLogGroup
               - logs:CreateLogStream
               - logs:PutLogEvents
-            Resource: 
-              !Join 
+            Resource: !Join
               - ""
               - - !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/"
-                - !Select [1, !Split [":function:", !GetAtt STMAnalyzeSegments.Arn]]
+                - !Select [
+                    1,
+                    !Split [
+                      ":function:",
+                      !GetAtt STMAnalyzeSegments.Arn,
+                    ],
+                  ]
                 - ":*"
 
-  STMAnalyzeSegmentsDataVersion:
-    Condition: SetSTMAnalyzeSegmentsData
+  STMAnalyzeSegmentsVersion:
+    Condition: STMAnalyzeSegmentsEnabled
     Type: AWS::Lambda::Version
-    DeletionPolicy: Retain
-    UpdateReplacePolicy: Retain
```

Comment:
Without the 2 lines with "Retain" as values, your function won't properly update through the versioning